tests/init/neg/inherit-non-hot.check

```diff
@@ -1,24 +1,17 @@
 -- Error: tests/init/neg/inherit-non-hot.scala:6:34 --------------------------------------------------------------------
 6 |            if b == null then b = new B(this) // error
   |                                  ^^^^^^^^^^^
-  |                        Cannot prove that the value is fully initialized. May only assign fully initialized value.
-  |                         Calling trace:
-  |                        -> val c = new C	[ inherit-non-hot.scala:19 ]
-  |                                   ^^^^^
-  |                        -> class C extends A {	[ inherit-non-hot.scala:15 ]
-  |                           ^
-  |                        -> val bAgain = toB.getBAgain	[ inherit-non-hot.scala:16 ]
-  |                                        ^^^
+  |                                  May only assign fully initialized value. Calling trace:
```

Comment:
1. The RHS of an assignment to a field that may be called from a constructor must be fully initialized.
2. This assignment may be called from a constructor because:
2a. The constructor of object Foo calls new C (line 19)
(It seems line 15 is not really relevant to the explanation.)
2b. The constructor of C calls A.toB (line 16)
2c. A.toB contains the assignment to the field.
3. I could not prove that the assigned value is fully initialized for the following reason.
3a. For the result of a constructor call to be fully initialized, all arguments of the call must be fully initialized. I could not prove that the argument this for the parameter a is fully initialized.
3b. I could not prove that this is fully initialized because the initializer of its field c has not yet run.